spatial-tracking-explainer.md

```diff
@@ -197,37 +197,58 @@ function onSessionStarted(session) {
+### Poses
+On a frame-by-frame basis, developers can query the location of any `XRSpace` within another `XRSpace` via the `XRFrame.getPose()` function. This function takes the `space` parameter which is the `XRSpace` to locate and the `relativeTo` parameter which defines the coordinate system in which the resulting `XRPose` should be returned. The `transform` attribute of `XRPose` is an `XRRigidTransform` representing the location of `space` within `relativeTo`. 
+
+While the `relativeTo` parameter is an `XRSpace`, developers will often choose to supply a `XRReferenceSpace` as the `relativeTo` parameter so that coordinates will be consistent with those used for rendering. For more information on rendering, see the main [WebXR explainer](explainer.md).
 
 ```js
-  let pose = frame.getPose(xrSpace, xrReferenceSpace);
+  let pose = xrFrame.getPose(xrSpace, xrReferenceSpace);
   if (pose) {
     // Do a thing
   }
 ```
 
-The `emulatedPosition` attribute of `XRPose` indicates that the translation components of retrieved pose matrices may not be accurate. There are a number of reasons this might be the case. For example, a headset with orientation-only tracking capability may include position data to represent neck modeling. Another reason might be the underlying platform's tracking-loss behavior causes orientation data to be updated while it is unable to update position data. In these situations, the `emulatedPosition` attribute will be set to `true`.
+#### Tracking loss
+Developers should check initially that the result from `getPose()` is not null, as the pose of `space` within `relativeTo` may not have been established yet. For example, a viewer may not yet have been tracked within the application's `XRReferenceSpace`, or a motion controller may not yet have been observed after the user turned it on.
 
-#### Rays
-An `XRRay` object includes both an `origin` and `direction`, both given as `DOMPointReadOnly`s. The `origin` represents a 3D coordinate in space with a `w` component that must be 1, and the `direction` represents a normalized 3D directional vector with a `w` component that must be 0. The `XRRay` also defines a `matrix` which represents the transform from a ray originating at `[0, 0, 0]` and extending down the negative Z axis to the ray described by the `XRRay`'s `origin` and `direction`. This is useful for positioning graphical representations of the ray.
+However, once a pose is initially established for a viewer or input source, pose matrices should continue to be provided even during tracking loss. The `emulatedPosition` attribute of `XRPose` indicates that the position component of the retrieved pose matrix does not represent an actively tracked position. There are a number of reasons this might be the case. For example:
```

Comment:
So far, all we have are reference spaces (including `viewer` spaces) and input source spaces.  As we later define anchor spaces and other spaces, we'd need to augment the spec if we decide they behave differently than today's spaces.

For now, I've just changed this to:
> However, once a pose is initially established for a space (e.g. a `viewer` reference space or an input source's `gripSpace`), pose matrices should continue to be provided even during tracking loss.

Reply:
My hesitation with this new phrasing is that pose data is always queried the same way (through `getPose()`) regardless of the type of `XRSpace` being located.  If future types inherited from `XRSpace` behave differently under tracking loss, we could end up in a situation where developers aren't prepared to handle those differences.  In other words, you're making the statement that the only time this should return `null` is during initialization and that once a pose is returned it will always be non-null, but I'm not sure we'd take that stance for anchors in the future.  Am I making sense?

Reply:
I'm not sure I can see why we'd ever return non-null;  assuming there is some way to know that the tracking has stopped, continuing to return the last know pose is the only option that makes sense to me.  If it's optional, we'll get a pattern where developers (and libraries) will just mirror that last known pose anyway, and use "null" as a "use last know pose" flag.

Reply:
There are two paths we may ultimately take with `XRSpace` tracking recovery for future spaces:
* We do as @blairmacintyre says and just decide that for all future spaces, we might as well just keep returning a last-known pose once any pose is delivered, and so all future spaces just behave as today's spaces do.
* We decide that it's simply too wrong to claim that we have any notion at all of where an `XRAnchor` is relative to the `viewer` reference space if you've recovered tracking in a disjoint new room, and so we want to be hardcore and return `null` for just that new flavor of `XRSpace`.  However, this decision won't surprise any existing apps - only apps that explicitly start using `XRAnchor` objects and their resulting spaces will ever observe the new behavior.  It seems OK for apps that explicitly opt in to a new scenario through new types to start getting more nuanced behavior from existing types, which we'd document at that point.

While I'm tempted toward the latter behavior to help stop apps from doing silly things when rendering anchored content, I believe the former is more consistent and easy to specify, and apps have the bit they need to not render at anchors when they're not actively tracked. (and as Blair points out, a naive abstraction layer above WebXR is likely to just implement "null" as "don't change the scene object's pose" anyway, and so there's a limit to how valuable being draconian is anyway)  However, whatever decision we do want to make later, I think we can do so compatibly, regardless of what we say now.